include/rmm/rmm.hpp

```diff
@@ -26,191 +26,181 @@ extern "C" {
-
-// RAII logger class
-class LogIt {
- public:
-  LogIt(Logger::MemEvent_t event, void* ptr, size_t size, cudaStream_t stream,
-        const char* filename, unsigned int line,
-        bool usageLogging = RMM_USAGE_LOGGING)
-      : event(event),
-        device(0),
-        ptr(ptr),
-        size(size),
-        stream(stream),
-        usageLogging(usageLogging),
-        line(line) {
-    if (filename) file = filename;
-    if (Manager::getOptions().enable_logging) {
-      cudaGetDevice(&device);
-      start = std::chrono::system_clock::now();
+    /// Sometimes you need to start logging before the pointer address is
+    /// known
+    inline void setPointer(void* p) {
+      if (Manager::getOptions().enable_logging) ptr = p;
     }
-  }
 
-  /// Sometimes you need to start logging before the pointer address is
-  /// known
-  inline void setPointer(void* p) {
-    if (Manager::getOptions().enable_logging) ptr = p;
-  }
+    ~LogIt() {
+      if (Manager::getOptions().enable_logging) {
+        Logger::TimePt end = std::chrono::system_clock::now();
+        size_t freeMem = 0, totalMem = 0;
+        if (usageLogging) rmmGetInfo(&freeMem, &totalMem, stream);
+        Manager::getLogger().record(event, device, ptr, start, end, freeMem,
+                                    totalMem, size, stream, file, line);
+      }
+    }
 
-  ~LogIt() {
-    if (Manager::getOptions().enable_logging) {
-      Logger::TimePt end = std::chrono::system_clock::now();
-      size_t freeMem = 0, totalMem = 0;
-      if (usageLogging) rmmGetInfo(&freeMem, &totalMem, stream);
-      Manager::getLogger().record(event, device, ptr, start, end, freeMem,
-                                  totalMem, size, stream, file, line);
+   private:
+    rmm::Logger::MemEvent_t event;
+    int device;
+    void* ptr;
+    size_t size;
+    cudaStream_t stream;
+    rmm::Logger::TimePt start;
+    std::string file;
+    unsigned int line;
+    bool usageLogging;
+  };
+
+  /**
+   * ---------------------------------------------------------------------------*
+   * @brief Allocate memory and return a pointer to device memory.
+   *
+   * @param[out] ptr Returned pointer
+   * @param[in] size The size in bytes of the allocated memory region
+   * @param[in] stream The stream in which to synchronize this command
+   * @param[in] file The filename location of the call to this function, for
+   * tracking
+   * @param[in] line The line number of the call to this function, for tracking
+   * @return rmmError_t RMM_SUCCESS, or RMM_ERROR_NOT_INITIALIZED if
+   * rmmInitialize has not been called, RMM_ERROR_INVALID_ARGUMENT if ptr is
+   *                    null, RMM_ERROR_OUT_OF_MEMORY if unable to allocate the
+   *                    requested size, or RMM_CUDA_ERROR on any other CUDA
+   * error.
+   * --------------------------------------------------------------------------**/
+  template <typename T>
+  inline rmmError_t alloc(T** ptr, size_t size, cudaStream_t stream,
```

Comment:
The macros used for error checking will now throw exceptions. 

I didn't endeavor to change the return type of `alloc/free/realloc` etc. as I assumed it would break some code relying on it to return error codes (such as a success error code).

Removing the error code return type certainly could be done, but is likely more involved.